// @gshell/mongodb/dist/mongodb.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const mongodb_1 = require("mongodb");
class GMongoDB {
    constructor(options) {
        this.options = options;
    }
    up() {
        return __awaiter(this, void 0, void 0, function* () {
            try {
                this.client = yield mongodb_1.MongoClient.connect(this.options.uri, { useNewUrlParser: true });
            }
            catch (err) {
                throw err; // TODO: Handle exception
            }
        });
    }
    down() {
        return __awaiter(this, void 0, void 0, function* () {
            try {
                this.client && (yield this.client.close());
            }
            catch (err) {
                throw err; // TODO: Handle exception
            }
        });
    }
    getDb(name) {
        try {
            return this.client.db(name ? name : this.options.name);
        }
        catch (err) {
            throw err;
        }
    }
}
exports.default = GMongoDB;
